### PR TITLE
SDL clean-up - remove HAS_SDL_OPENGL

### PR DESCRIPTION
**Pull request by FernetMenta**
see title

let's get rid of that crap completely

### CODE REVIEW

**Comment by FernetMenta** [comment]
jenkins build this please

File: xbmc/Application.cpp [hunk 1]
```diff
@@ -807,7 +807,7 @@ bool CApplication::CreateGUI()
 
   uint32_t sdlFlags = 0;
 
-#if (defined(HAS_SDL_OPENGL) || (HAS_GLES == 2)) && !defined(HAS_GLX)
+#if defined(TARGET_DARWIN_OSX)
```
Comment (FernetMenta) [inline xbmc/Application.cpp]: @Memphiz could you check this please. I don't think any or our GLES platforms other than iOS could have made use of this

Comment (Memphiz) [inline xbmc/Application.cpp]: Ios doesn't use sdl at all - Looks good

**Comment by FernetMenta** [comment]
jenkins build this please

**Pull request merged**